linux-glibc-2.17-x64/Dockerfile

```diff
@@ -0,0 +1,118 @@
+FROM ubuntu:24.04
+
+ARG GO_VERSION
+ARG GO_SHA256_LINUX_AMD64
+ARG CTNG_VERSION=1.26.0
+ARG IBM_MQ_VERSION=9.2.4.0
+ARG IBM_MQ_SHA256="d0d583eba72daf20b3762976f8831c2e23150ace90509520e12f8cda5b5bdb49"
+ARG RUST_VERSION=1.76.0
+ARG RUSTC_SHA256="0b2f6c8f85a3d02fde2efc0ced4657869d73fccfce59defb4e8d29233116e6db"
+ARG RUSTUP_VERSION=1.26.0
+ARG RUSTUP_SHA256="0b2f6c8f85a3d02fde2efc0ced4657869d73fccfce59defb4e8d29233116e6db"
+ARG BUNDLER_VERSION=2.4.20
+ARG DD_TARGET_ARCH
+
+ENV CONDA_PATH /root/miniforge3
+
+# Remove the early return on non-interactive shells, which makes sourcing the file not activate conda
+RUN grep -v return /root/.bashrc >> /root/newbashrc && cp /root/newbashrc /root/.bashrc
+
+RUN apt update -qy && apt install -y \
+    wget xz-utils gpg build-essential flex texinfo unzip \
+    help2man file gawk libtool-bin bison libncurses-dev \
+    python-is-python3 git cmake curl fakeroot procps bzip2 \
+    pkg-config libssl-dev libcurl4-openssl-dev libexpat-dev libpq-dev libz-dev \
+    rpm tar gettext autopoint autoconf clang libtool-bin \
+    pkg-config flex meson selinux-basics squashfs-tools gpg xz-utils gnupg2 patchelf cpio
+
+COPY linux-glibc-2.17-x64/config-x86_64-unknown-gnu-linux-glibc2.17 /build/crosstool-ng-${CTNG_VERSION}/.config
+COPY linux-glibc-2.17-x64/toolchain.cmake /opt/cmake/x86_64-unknown-linux-gnu.toolchain.cmake
+COPY linux-glibc-2.17-x64/cargo-config.toml /root/.cargo/config.toml
+
+RUN git config --global user.email "package@datadoghq.com"
+RUN git config --global user.name "Bits"
+
+# Build texinfo to get a bugfix version
+# Otherwise generating glibc doc with a recent version
+# causes a segfault. The fix isn't yet released as a package
+# https://git.savannah.gnu.org/cgit/texinfo.git/commit/?id=60d3edc4b74b4e1e5ef55e53de394d3b65506c47
+RUN cd /build && \
+    git clone https://git.savannah.gnu.org/git/texinfo.git && \
+    cd texinfo && \
+    git reset --hard 60d3edc4b74b4e1e5ef55e53de394d3b65506c47 && \
+    ./autogen.sh && \
+    ./configure && make -j$(ncproc) && make install && \
+    rm -rf /build/texinfo
+
+RUN cd /build && \
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏ are these not available at `https` addresses?

Reply:
They are not, or not in a reliable fashion. Currently it times out when I thy to access that URL through HTTPS.

You might want to ask what's the point of checking a GPG signature that's downloaded over HTTP, and you'd be right to, however I have no answer :(

Reply:
Weird - but fine

Reply:
Heh I guess that it's not possible to forge a signature file that would still match the private key, and the key itself is delivered over a secured channel

Reply:
For the sake of not having to use HTTP, could we use the artifacts from GitHub releases instead?

They provide the `.tar.xz` and the signature file there as well: https://github.com/crosstool-ng/crosstool-ng/releases/tag/crosstool-ng-1.26.0

Reply:
Very good point, done!